entry/src/entry.rs

```diff
@@ -524,46 +524,48 @@ fn start_verify_transactions_gpu(
         });
     }
 
-    let mut packet_batches = entry_txs
-        .par_iter()
-        .chunks(PACKETS_PER_BATCH)
-        .map(|slice| {
-            let vec_size = slice.len();
-            let mut packet_batch = PacketBatch::new_with_recycler(
-                &verify_recyclers.packet_recycler,
-                vec_size,
-                "entry-sig-verify",
-            );
-            // We use set_len here instead of resize(vec_size, Packet::default()), to save
-            // memory bandwidth and avoid writing a large amount of data that will be overwritten
-            // soon afterwards. As well, Packet::default() actually leaves the packet data
-            // uninitialized, so the initialization would simply write junk into
-            // the vector anyway.
-            unsafe {
-                packet_batch.set_len(vec_size);
-            }
-            let entry_tx_iter = slice
-                .into_par_iter()
-                .map(|tx| tx.to_versioned_transaction());
-
-            let res = packet_batch
-                .par_iter_mut()
-                .zip(entry_tx_iter)
-                .all(|(packet, tx)| {
-                    *packet.meta_mut() = Meta::default();
-                    Packet::populate_packet(packet, None, &tx).is_ok()
-                });
-            if res {
-                Ok(packet_batch)
-            } else {
-                Err(TransactionError::SanitizeFailure)
-            }
-        })
-        .collect::<Result<Vec<_>>>()?;
+    let packet_batches = thread_pool.install(|| {
+        transactions
+            .par_chunks(PACKETS_PER_BATCH)
```

Comment:
Any idea how many transactions we tend to have coming through here? `PACKETS_PER_BATCH` is 64, so I wonder how many threads we're spinning up typically. If it's sufficiently small, maybe the nested parallelism was helping us chew through these 64 transactions in a batch.

I still like your idea of having a single level to divide and conquer, but I'm looking for any potential changes to current behavior that could cause performance side effects

Reply:
> but I'm looking for any potential changes to current behavior that could cause performance side effects

Definitely, I appreciate the thoughtful review!

We have a safeguard higher up to ensure we only try to use the GPU if we have at least 512 packets:
https://github.com/anza-xyz/agave/blob/7640c25509b0aa299b7e3238208c1d01035e7898/entry/src/entry.rs#L432-L455

Reply:
Got it, so at least 8 threads in the mix. That sounds very reasonable